python/lsst/metaserv/schemaToMeta.py

```diff
@@ -0,0 +1,271 @@
+    extra tokens in comments. Extracts information for each table:
+    * name
+    * engine
+    * per column:
+      - type
+      - notnull
+      - defaultValue
+      - <descr>...</descr>
+      - <unit>...</unit>
+      - <ucd>...</ucd>
+    and saves it in an array.
+
+    This code was originally written for schema browser
+    (in cat/bin/schema_to_metadata.py).
+    """
+
+    def __init__(self, inputFileName):
+        """
+        Constructor, prepares for execution.
+
+        @param inputFileName:   ASCII file containing the schem to be parsed
+        """
+        if not os.path.isfile(inputFileName):
+            sys.stderr.write("File '%s' does not exist\n" % inputFileName)
+            sys.exit(1)
+        self._inFName = inputFileName
+
+
+    def parse(self):
+        """Do actual parsing. Returns the retrieved structure as a table."""
+        in_table = None
+        in_col = None
+        in_colDescr = None
+        table = {}
+
+        tableStart = re.compile(r'CREATE TABLE (\w+)*')
+        tableEnd = re.compile(r"\)")
+        engineLine = re.compile(r'\) (ENGINE|TYPE)=(\w+)*;')
+        columnLine = re.compile(r'[\s]+(\w+) ([\w\(\)]+)')
+        descrStart = re.compile(r'<descr>')
+        descrEnd = re.compile(r'</descr>')
+        unitStart = re.compile(r'<unit>')
+        unitEnd = re.compile(r'</unit>')
+
+        colNum = 1
+
+        iF = open(self._inFName, mode='r')
+        for line in iF:
+            # print "processing ", line
+            m = tableStart.search(line)
+            if m is not None:
+                tableName = m.group(1)
+                table[tableName] = {}
+                colNum = 1
+                in_table = table[tableName]
+                in_col = None
+                #print "Found table ", in_table
+            elif tableEnd.match(line):
+                m = engineLine.match(line)
+                if m is not None:
+                    engineName = m.group(2)
+                    in_table["engine"] = engineName
+                # print "end of the table"
+                # print in_table
+                in_table = None
+            elif in_table is not None: # process columns for given table
+                m = columnLine.match(line)
+                if m is not None:
+                    firstWord = m.group(1)
+                    if self._isIndexDefinition(firstWord):
+                        t = "-"
+                        if firstWord == "PRIMARY":
+                            t = "PRIMARY KEY"
+                        elif firstWord == "UNIQUE":
+                            t = "UNIQUE"
+                        idxInfo = {"type" : t,
+                                   "columns" : self._retrColumns(line)
+                               }
+                        if "indexes" not in in_table:
+                            in_table["indexes"] = []
+                        in_table["indexes"].append(idxInfo)
```

Comment:
Three above lines can be be done as:
`in_table.setdefault("indexes", []).append(idxInfo)`